### PR TITLE
Document release process

### PR DESCRIPTION
**Pull request by pellared**
## Why

> Why this pull request was created? Explain the problem that this pull request is trying to solve.

Fixes https://github.com/goyek/goyek/issues/53

> Apart from that: would we have a checklist for doing a release? How would I know "all" is included? ;)

## What

> What does this pull request contain? Explain what is this pull request changing.

- Automate updating pkg.go.dev
- Add `RELEASING.md`

## Checklist

- [x] `CHANGELOG.md` is updated. N/A
- [ ] `README.md` is updated. _not sure if it is worth to add anything_
- [x] The code changes follow [Effective Go](https://golang.org/doc/effective_go).

### CODE REVIEW

File: RELEASING.md [hunk 1]
```diff
@@ -0,0 +1,17 @@
+# Releasing
+
+## Pre-release
+
+Create a pull request named `Release v<version>` that does the following:
+
+1. Update `go.mod` for submodules to depend on the new release.
```
Comment (pellared) [inline RELEASING.md:7]: I want to make a submodule in `./build` like here: https://github.com/pellared/fluentassert/tree/main/build
Thanks to it we can get rid of `tools` directory and e.g. use `go-shellwords` in our build pipeline

**Comment by dertseha** [comment]
Thank you for starting this. Now I would know which files to touch, but not what exactly to do. When you write "Update xyz" - what does "updating" include?
As an example, at least for changelog I believe it means updating the current header with a fixed date (?) and preparing a new header... right? Other example: readme might include a warning of changed API, this header should be removed.
It's possible that not "all" eventualities can be covered, yet the clear items should be listed.

It does not have to be a markdown checklist - it's enough to have a list of items to do. Think of all the things you believe are "obvious" and critically think if a new person to the project would know that. Different view: If you yourself were stressed, which things would you potentially skip?

**Review by dertseha — APPROVED** [review]
Thank you.
How about I perform the next release, to try it out?

**Comment by pellared** [comment]
> Thank you.
> How about I perform the next release, to try it out?

I want to do this: https://github.com/goyek/goyek/pull/83#discussion_r637290179 before release. I will try to make a PR tonight.

Afterward, I think it would be great if you do the release 👍

**Comment by pellared** [comment]
@dertseha Changed my mind. Go for it 😄 This https://github.com/goyek/goyek/pull/83#discussion_r637290179 is very internal.

**Pull request merged**